rskj-core/src/main/java/org/ethereum/rpc/exception/RskErrorResolver.java

```diff
@@ -1,82 +1,75 @@
+    }
 
     @Override
-    public JsonError resolveError(Throwable t, Method method, List<JsonNode> arguments) {
-        JsonError error;
+    public JsonError resolveError(Throwable throwable, Method method, List<JsonNode> arguments) {
+        if (throwable instanceof JsonMappingException && throwable.getMessage() != null && throwable.getMessage().contains("Can not construct instance")) {
+            return new JsonMappingExceptionHandler().apply(throwable, method, arguments);
+        }
 
-        if (t instanceof InvalidRskAddressException) {
-            error = new JsonError(
-                    JsonRpcError.INVALID_PARAMS,
-                    "invalid argument 0: hex string has length " + arguments.get(0).asText().replace("0x", "").length() + ", want 40 for RSK address",
-                    null);
-        } else if (t instanceof RskJsonRpcRequestException) {
-            error = new JsonError(((RskJsonRpcRequestException) t).getCode(), t.getMessage(), null);
-        } else if (t instanceof InvalidFormatException) {
-            error = new JsonError(JsonRpcError.INTERNAL_ERROR, "Internal server error, probably due to invalid parameter type", null);
-        } else if (t instanceof UnrecognizedPropertyException) {
-            error = new JsonError(
-                    JsonRpcError.INVALID_PARAMS,
-                    getExceptionMessage((UnrecognizedPropertyException) t),
-                    null);
-        } else if (t instanceof JsonMappingException && t.getMessage().contains("Can not construct instance")) {
-            error = new JsonError(
-                    JsonRpcError.INVALID_PARAMS,
-                    "invalid argument 0: json: cannot unmarshal string into value of input",
-                    null);
-        } else if (t instanceof UnsupportedOperationException || (t.getMessage() != null && t.getMessage().toLowerCase().contains("method not supported"))) {
-            error = new JsonError(
-                    JsonRpcError.METHOD_NOT_FOUND,
-                    "the method " + method.getName() + " does not exist/is not available",
-                    null);
-        } else {
-            logger.error("JsonRPC error when for method {} with arguments {}", method, arguments, t);
-            error = new JsonError(JsonRpcError.INTERNAL_ERROR, "Internal server error", null);
+        if (throwable.getMessage() != null && throwable.getMessage().toLowerCase().contains("method not supported")) {
+            return new UnsupportedOperationExceptionHandler().apply(throwable, method, arguments);
         }
-        return error;
-    }
 
-    private static String getExceptionMessage(UnrecognizedPropertyException ex) {
-        if (ex.getPropertyName() == null || ex.getKnownPropertyIds() == null) return "Invalid parameters";
+        Class<? extends RskErrorHandler> handlerClass = errorHandlers.get(throwable.getClass());
```

Comment:
I guess we're missing here a case when some derived exceptions are thrown. For instance, there could be a thrown exception derived from `InvalidFormatException`, so that this mapping won't return a result. Probably better to stick to the `else if (t instanceof InvalidFormatException)` construction, wdyt?

Reply:
done (reverted)